### PR TITLE
✨ Add IR remote entity, enhance config flow, improve tests & CI

### PR DESCRIPTION
**Pull request by MapleEve**
#### 💻 变更类型 | Change Type

-  [x] ✨ feat
- 🐛 fix
- ♻️ refactor
- 💄 style
- 👷 build
- ⚡️ perf
- 📝 docs
- 🔨 chore

#### 🔀 变更说明 | Description of Change

1 优化CI构建和测试
2 增加测试代码覆盖率
3 优化红外相关服务

## Sourcery 总结

提高了服务和传感器的健壮性，添加了红外遥控器实体支持和遥控器的多步配置流程 (config_flow)，扩展了测试以覆盖边缘情况和兼容性，强制执行 88 字符的代码风格，并通过 `pytest-homeassistant-custom-component` 插件简化了 CI 依赖管理。

新功能:
- 添加了带有 `LifeSmartRemoteEntity` 的红外遥控器实体平台
- 增强了 `config_flow` 以支持红外遥控器的多步配置（类别、品牌、型号）
- 引入了 `OpenAPIClient` 方法 `get_ir_custom_keys_async` 和 `get_ir_remote_feature_async`

Bug 修复:
- 确保 `send_ir_keys` 和 `send_ackeys` 服务需要 `ai` 或 `idx` 之一，并正确处理 `keyDetail`
- 修复了 `CO2` 值以及未知设备/单位/状态类的传感器转换逻辑
- 修正了 `SL_NATURE P5` 的传感器 `device_class` 解析和气候全局刷新处理
- 在 `light.async_turn_on` 中将色温限制在最小/最大范围内，并添加了回退效果逻辑

改进:
- 更新代码风格为 `Black/isort` 行长 88，并简化格式化配置
- 在传感器、灯、集线器、兼容性和 `client_base` 模块中添加了大量的断言消息和数百个新测试
- 重构 CI，仅安装 `pytest-homeassistant-custom-component` 插件并自动管理 `pytest`/HA 版本

CI:
- 简化 GitHub Actions 和本地测试脚本，使其依赖于 `pytest-homeassistant-custom-component` 插件

文档:
- 优化 `services.yaml`，增加选择器、必填标志和详细的字段文档
- 更新 `README` 和 `Chinese README` 以反映新的风格指南和版本信息

杂项:
- 将集成版本提升至 `v25.08.10`

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Improve robustness of services and sensors, add IR remote entity support and multi-step config_flow for remotes, expand tests to cover edge cases and compatibility, enforce 88-char code style, and simplify CI dependency management via pytest-homeassistant-custom-component plugin.

New Features:
- Added remote entity platform with LifeSmartRemoteEntity for IR remotes
- Enhanced config_flow to support multi-step configuration of IR remotes (category, brand, model)
- Introduced OpenAPIClient methods get_ir_custom_keys_async and get_ir_remote_feature_async

Bug Fixes:
- Ensure send_ir_keys and send_ackeys services require either ai or idx and correctly handle keyDetail
- Fix sensor conversion logic for CO2 values and unknown device/unit/state classes
- Correct sensor device_class resolution for SL_NATURE P5 and climate global refresh handling
- Clamp color temperature within min/max and fallback effect logic in light.async_turn_on

Enhancements:
- Update code style to Black/isort line length 88 and simplify formatting configs
- Add extensive assertion messages and hundreds of new tests across sensors, lights, hub, compatibility, and client_base modules
- Refactor CI to install only pytest-homeassistant-custom-component plugin and manage pytest/HA versions automatically

CI:
- Simplify GitHub Actions and local test scripts to rely on pytest-homeassistant-custom-component plugin

Documentation:
- Refine services.yaml with selectors, required flags, and detailed field documentation
- Update README and Chinese README to reflect new style guidelines and version info

Chores:
- Bump integration version to v25.08.10

</details>

### CODE REVIEW

**Comment by MapleEve** [comment]
👍 @MapleEve

Thank you for raising your pull request and contributing to LifeSmart integration.
Please make sure you have followed our contributing guidelines. We will review it as soon as possible.
If you encounter any problems, please feel free to connect with us.\
非常感谢您提出拉取请求并为 LifeSmart 集成做出贡献，请确保您已经遵循了我们的贡献指南，我们会尽快审查它。
如果您遇到任何问题，请随时与我们联系。

File: custom_components/lifesmart/tests/test_openapi_client.py [hunk 1]
```diff
@@ -480,14 +486,14 @@ class TestAPIWrapperMethods:
             (
                 "async_send_ir_key",
                 "SendKeys",
-                ("hub1", "ai1", "me1", "cat1", "brand1", "key1"),
+                ("hub1", "me1", "cat1", "brand1", "key1", "ai1"),
```
[the remark this answers is not in the document]
Comment (MapleEve) [reply custom_components/lifesmart/tests/test_openapi_client.py:489]: 已经处理

File: custom_components/lifesmart/tests/test_sensor.py [hunk 2]
```diff
@@ -851,6 +863,339 @@ async def test_sensor_availability_restoration(
...
+            raw_device=unknown_unit_device,
+            client=MagicMock(),
+            entry_id="test",
+            sub_device_key="UNKNOWN_UNIT",
+            sub_device_data={"val": 50, "type": 1},
+        )
+
+        # 未知单位应该返回None
+        assert sensor.native_unit_of_measurement is None, "未知单位应该返回None"
+
```
[the remark this answers is not in the document]
Comment (MapleEve) [reply custom_components/lifesmart/tests/test_sensor.py:921]: 1. test_sensor.py
    - ✅ 删除了重复的 test_sensor_entity_unknown_device_class (第二个副本在行1498)
    - ✅ 删除了重复的 test_sensor_entity_unknown_unit_measurement (第二个副本在行1526)
    - ✅ 删除了重复的 test_sensor_entity_unknown_state_class (第二个副本在行1554)
  2. test_local_tcp_client.py
    - ✅ 删除了重复的 test_check_login_connection_closed_by_peer (第二个副本在行1292)
    - ✅ 删除了重复的 test_check_login_eof_error_handling (第二个副本在行1316)
    - ✅ 删除了重复的 test_check_login_writer_close_error_handling (第二个副本在行1357)

File: custom_components/lifesmart/tests/test_sensor.py [hunk 3]
```diff
@@ -851,6 +863,339 @@ async def test_sensor_availability_restoration(
...
+            raw_device=unknown_device,
+            client=MagicMock(),
+            entry_id="test",
+            sub_device_key="UNKNOWN_KEY",
+            sub_device_data={"val": 100, "type": 1},
+        )
+
+        # 未知设备类别应该返回None
+        assert sensor.device_class is None, "未知设备类别应该返回None"
+
```
[the remark this answers is not in the document]
Comment (MapleEve) [reply custom_components/lifesmart/tests/test_sensor.py:893]: 1. test_sensor.py
    - ✅ 删除了重复的 test_sensor_entity_unknown_device_class (第二个副本在行1498)
    - ✅ 删除了重复的 test_sensor_entity_unknown_unit_measurement (第二个副本在行1526)
    - ✅ 删除了重复的 test_sensor_entity_unknown_state_class (第二个副本在行1554)
  2. test_local_tcp_client.py
    - ✅ 删除了重复的 test_check_login_connection_closed_by_peer (第二个副本在行1292)
    - ✅ 删除了重复的 test_check_login_eof_error_handling (第二个副本在行1316)
    - ✅ 删除了重复的 test_check_login_writer_close_error_handling (第二个副本在行1357)

**Comment by MapleEve** [comment]
@sourcery-ai resolve

**Comment by MapleEve** [comment]
@sourcery-ai guide
